quadkey/library/test/quadkey_test.js

```diff
@@ -0,0 +1,140 @@
+const fs = require('fs');
+const assert = require('assert').strict;
+/* Emulate how BigQuery would load the file */
+global.eval(fs.readFileSync('../../quadkey_library.js') + '');
+
+describe('QUADKEY unit tests', () => {
+    it('Version', async() => {
+        assert.equal(quadkeyVersion(), 1);
+    });
+
+    it('Should be able to encode/decode tiles at any level of zoom', async() => {
+        let tilesPerLevel, x, y, xDecoded, yDecoded, zDecoded;
+        for (let z = 0; z < 30; ++z) {
+            if (z === 0) {
+                tilesPerLevel = 1;
+            } else {
+                tilesPerLevel = 2 << (z - 1);
+            }
+
+            x = 0;
+            y = 0;
+            let zxyDecoded = ZXYFromQuadint(quadintFromZXY(z, x, y));
+            zDecoded = zxyDecoded.z;
+            xDecoded = zxyDecoded.x;
+            yDecoded = zxyDecoded.y;
+            assert.ok(z === zDecoded && x === xDecoded && y === yDecoded);
+
+            if (z > 0) {
+                x = tilesPerLevel / 2;
+                y = tilesPerLevel / 2;
+                zxyDecoded = ZXYFromQuadint(quadintFromZXY(z, x, y));
+                zDecoded = zxyDecoded.z;
+                xDecoded = zxyDecoded.x;
+                yDecoded = zxyDecoded.y;
+                assert.ok(z === zDecoded && x === xDecoded && y === yDecoded);
+
+                x = tilesPerLevel - 1;
+                y = tilesPerLevel - 1;
+                zxyDecoded = ZXYFromQuadint(quadintFromZXY(z, x, y));
+                zDecoded = zxyDecoded.z;
+                xDecoded = zxyDecoded.x;
+                yDecoded = zxyDecoded.y;
+                assert.ok(z === zDecoded && x === xDecoded && y === yDecoded);
+            }
+        }
+    });
+
+    it('Should be able to encode/decode between quadint and quadkey at any level of zoom', async() => {
```

Comment:
I think I would add some known value tests and not only rountrip, otherwise you risk breaking compatibility and not noticing it.